concordium-node/src/bin/cli.rs

```diff
@@ -625,21 +625,27 @@ async fn import_missing_blocks(
         index_response.status().canonical_reason().unwrap()
     );
 
-    let mut index_reader = index_response
-        .bytes_stream()
-        .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))
-        .into_async_read();
-    let mut first_line: String = String::new();
-    index_reader.read_line(&mut first_line).await?;
+    let index_str = index_response
+        .text()
+        .await
+        .context("Unable to get the catchup index file response text.")?;
+
+    let mut reader = csv::ReaderBuilder::new().has_headers(false).from_reader(index_str.as_bytes());
+    let first_line = reader.records().next().context(
+        "The catchup index file was empty. Please verify that you specified a correct catchup \
+         service URL. If the specified URL is correct, contact the catchup service administrator.",
+    )??;
```

Comment:
I recommend adding something like
```
.context("Could not parse the index file as a CSV file.")
```

Reply:
The return type of next() is `Option<Result<T>>`. Simply adding one ? would yield `Result<T>`. The context returns a Result<Result<T>> which seems to be Error in case of None. I guess the inner `.context` should suffice since I suppose it propagates outwards (but perhaps I am wrong). I tested it by providing it with an empty file and the correct error message is displayed.